src/sentry/integrations/message_builder.py

```diff
@@ -139,9 +143,29 @@ def build_footer(
     footer = f"{group.qualified_short_id}"
     if rules:
         rule_url = build_rule_url(rules[0], group, project)
-        footer += f" via {url_format.format(text=rules[0].label, url=rule_url)}"
+        # If this notification is triggered via the "Send Test Notification"
+        # button then the label is not defined, but the url works.
+        text = rules[0].label if rules[0].label else "Test Alert"
+        footer += f" via {url_format.format(text=text, url=rule_url)}"
 
         if len(rules) > 1:
             footer += f" (+{len(rules) - 1} other)"
 
     return footer
+
+
+def has_releases(project: Project) -> bool:
```

Comment:
I think we have a project flag for this `project.flags.has_releases` and it was backfilled [here](https://github.com/getsentry/sentry/blob/f9f07c369f43cc40ee82e9c995f5405962fd0a67/src/sentry/migrations/0290_fix_project_has_releases.py#L9). I'd double check that it's still in use, but this should do what we want